src/components/Mandal/view/SubMandal.js

```diff
@@ -1,40 +1,45 @@
 import React from "react";
 import { useDispatch, useSelector } from "react-redux";
 import styled from "styled-components";
+import PropTypes from "prop-types";
 
 import MandalBox from "./MandalBox";
-import { displayMain } from "../../../features/viewSlice";
-
+import { displayMain, displaySub } from "../../../features/viewSlice";
+import { showBoxes } from "./utils";
 const BoxContainer = styled.div`
   display: grid;
-  height: 570px;
-  width: 570px;
+  height: 100%;
+  width: 100%;
   grid-template-columns: 1fr 1fr 1fr;
+  grid-template-rows: 1fr 1fr 1fr;
+  & :nth-child(n) {
+    border: none;
+  }
 `;
 
-export default function SubMandal() {
-  const data = useSelector(state => state.view.displayed);
+export default function SubMandal({ data, mandalIndex }) {
   const dispatch = useDispatch();
+  const viewOption = useSelector(state => state.view.option);
+
+  const handleBoxClick = (event, index) => {
+    if (viewOption === "full") {
+      dispatch(displaySub(mandalIndex));
+      return;
+    }
 
-  const handleBoxClick = index => {
     if (index === 4) {
       dispatch(displayMain());
     }
   };
 
-  const showBoxes = () => {
-    return data.map((box, index) => {
-      return (
-        <MandalBox
-          context={String(box.title)}
-          role={box.role}
-          key={box["_id"]}
-          goalId={box["_id"]}
-          onClick={() => handleBoxClick(index)}
-        />
-      );
-    });
-  };
-
-  return <BoxContainer className="gridContainer">{showBoxes()}</BoxContainer>;
+  return (
+    <BoxContainer className="gridContainer">
+      {showBoxes(data, handleBoxClick)}
+    </BoxContainer>
+  );
 }
+
+SubMandal.propTypes = {
+  data: PropTypes.instanceOf(Array),
```

Comment:
위에서 쓴 것과 동일한 맥락의 코멘트입니다!
```suggestion
  data: PropTypes.array,
```